### PR TITLE
SONARJAVA-2031 Filter out object methods from SAM lookup

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java [hunk 1]
```diff
@@ -1468,4 +1467,10 @@ public void getClass_return_type() {
     assertThat(((WildCardType) param).bound.is("ISuiteListener")).isTrue();
   }
 
+  @Test
+  public void lambda_on_sam_interface_defining_object_methods() throws Exception {
+    Result result = Result.createFor("SAMResolution");
+    assertThat(((VariableTree) result.symbol("var").declaration()).initializer().symbolType().is("MyVar"));
```
Comment (SonarTech) [inline java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java]: BLOCKER Complete the assertion. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2970)

File: java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java [hunk 2]
```diff
@@ -1468,4 +1467,10 @@ public void getClass_return_type() {
...
+    assertThat(((VariableTree) result.symbol("var2").declaration()).initializer().symbolType().is("MyVar2"));
```
Comment (SonarTech) [inline java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java]: BLOCKER Complete the assertion. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2970)

File: java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java [hunk 3]
```diff
@@ -1001,6 +1005,16 @@ private static JavaType getPrimitive(JavaType primitiveOrWrapper) {
     return Optional.empty();
   }
 
+  private boolean isObjectMethod(JavaSymbol.MethodJavaSymbol methodJavaSymbol) {
+    JavaSymbol.MethodJavaSymbol overriddenSymbol = methodJavaSymbol.overriddenSymbol();
+    boolean isObjectMethod = false;
+    while (overriddenSymbol != null && !isObjectMethod) {
+      isObjectMethod = overriddenSymbol.owner.type == symbols.objectType;
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java]: MAJOR Change this comparison to use the equals method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1698)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* BLOCKER 2 blocker
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java (hunk 3 above)
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java]: nope, I know what i'm doing, thanks

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**